src/quacc/settings.py

```diff
@@ -569,3 +570,36 @@ def change_settings(changes: dict[str, Any]):
     finally:
         for attr, original_value in original_values.items():
             setattr(SETTINGS, attr, original_value)
+
+
+def change_settings_wrap(func: Callable, changes: dict[str, Any]) -> Callable:
+    """
+    Wraps a function with the change_settings context manager if not already wrapped.
+
+    Parameters
+    ----------
+    func
+        The function to wrap.
+    changes
+        The settings to apply within the context manager.
+
+    Returns
+    -------
+    Callable
+        The wrapped function.
+    """
+    from quacc import change_settings
+
+    if hasattr(func, "__changed__") and func.__changed__:
+        original_func = func.original_func
+    else:
+        original_func = func
```

Comment:
With the knowledge that `.__wrapped__` is always there with `@wraps`, this logical can be simplified to something like:

```python
original_func = func.__wrapped__ if getattr(func, "__changed__", False) else func
```